### DISCUSSION TITLE
People Search : Issue with Name containing single quotes/apostrophe

### DISCUSSION THREAD

**Discussion by siva111** [microsoft-search/pnp-modern-search#3423] [Q&A]
**Version used**
4.8

**Describe the bug**
While searching for People users who having single quotes or apostrophe in Preferred Name/Last Name are not returned in results properly. Especially if we enter keyword ending with single quotes.
Even mapped Preferred name cp into RefinableStringLn00 and tried with it. same result

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Configure for People Search
2. Type in keyword of name till single quotes in last name , for eg Name: Mark L'Heureux
3. If we type in Mark L*    --> returns the user
4. If we type in Mark L'*  --> user not returned ( single quotes before * )
5. Is we type in Mark L'H*  --> returns the user
6. If we keep typing rest of words in Mark L'Heureux  --> returns the users
7. So if we search with single quotes at end its not returning result.

**Expected behavior**
Name should be searched partially with letters present in the name.

**Screenshots**

**Desktop (please complete the following information):**

**Additional context**
In OOTB Search box if we type in till single quotes it gives back result.

**Comment by kasperbolarsen** [comment]
I can confirm your finding, however it very much seems to be an API issue. I tested the various search query options using PnP.PowerShell ( and the SP Editor) and either leaving the single quote out or escaping it using ` seems to be the only variant that works.
Perhaps we can get it to work once Microsoft completes the Graph people Search but it is still in beta.

$res2 = submit-pnpsearchquery -Query "mark l*" -SourceId "b09a7990-05ea-4af9-81ef-edfab16c4e31" -Connection $spconn 
$res2.ResultRows.count
$res3 = submit-pnpsearchquery -Query "mark l'*" -SourceId "b09a7990-05ea-4af9-81ef-edfab16c4e31" -Connection $spconn
$res3.ResultRows.count
$res4 = submit-pnpsearchquery -Query "mark l'H*" -SourceId "b09a7990-05ea-4af9-81ef-edfab16c4e31" -Connection $spconn
$res4.ResultRows.count
$res5 = submit-pnpsearchquery -Query "mark l`'H*" -SourceId "b09a7990-05ea-4af9-81ef-edfab16c4e31" -Connection $spconn
$res5.ResultRows.count

**Comment by wobba** [comment]
This is an API issue and most likely ' is tokenization/normalized away on indexing and searching.
